[ews-build.webkit.org] Use Twisted defer for reviewer queries

JonWBedard opened the pull request:
#### 9b7933872a4ab7bc6617fd8b7ecd700331ac877a
<pre>
[ews-build.webkit.org] Use Twisted defer for reviewer queries
<a href="https://bugs.webkit.org/show_bug.cgi?id=250765">https://bugs.webkit.org/show_bug.cgi?id=250765</a>
rdar://104382329

Reviewed by Aakash Jain.

Replace fetch_data_from_url_with_authentication_github with a Twisted
version of the same function, but keep around the old function while
we transition. Adopt the Twisted version of the function when listing
reviewers of a PR.

* Tools/CISupport/ews-build/steps.py:
(GitHubMixin):
(GitHubMixin.fetch_data_from_url_with_authentication_github): Convert to Twisted.
(GitHubMixin.fetch_data_from_url_with_authentication_github_old): Retain old
synchronous version.
(GitHubMixin.get_pr_json): Use old synchronous version.
(GitHubMixin.get_reviewers): Use new Twisted version.
(GitHubMixin.remove_labels): Use old synchronous version.
(ValidateCommitterAndReviewer.run): Use Twisted asynchronous requests.
* Tools/CISupport/ews-build/steps_unittest.py:
(TestGitHubMixin.test_no_reviewers): Return defered to match API we&apos;re mocking.
(TestGitHubMixin.test_single_review): Ditto.
(TestGitHubMixin.test_multipe_reviews): Ditto.
(TestGitHubMixin.test_retracted_review): Ditto.
(TestGitHubMixin.test_pagination): Ditto.
(TestGitHubMixin.test_reviewers_invalid_response): Ditto.
(TestGitHubMixin.test_reviewers_error): Ditto.

Canonical link: <a href="https://commits.webkit.org/259427@main">https://commits.webkit.org/259427@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/585dfcc89c078666fded86c7ff470aa03c22a01a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104898 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13978 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37793 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/114165 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/174355 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/108803 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/15110 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4902 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/97224 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/113187 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/110658 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/11674 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/94682 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/39193 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/93543 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/26303 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/80855 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/7322 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/27662 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/7415 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/4252 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/103700 "Passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/13473 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/47213 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/9207 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3450 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -171,7 +171,26 @@ class GitHubMixin(object):
+        else:
+            defer.returnValue(response)
+
+    def fetch_data_from_url_with_authentication_github_old(self, url):
```
Maybe add a comment here to indicate reason to keep this or when to delete this

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -171,7 +171,26 @@ class GitHubMixin(object):
+            logger=self._addToLog,
+        )
+        if response and response.status_code // 100 != 2:
+            self._addToLog('stdio', 'Accessed {url} with unexpected status code {status_code}.\n'.format(url=url, status_code=response.status_code))
```
Can use Python 3 f-string here.

aj062 reviewed (approved):
r+ assuming this is tested.

JonWBedard committed:
[ews-build.webkit.org] Use Twisted defer for reviewer queries
https://bugs.webkit.org/show_bug.cgi?id=250765
rdar://104382329

Reviewed by Aakash Jain.

Replace fetch_data_from_url_with_authentication_github with a Twisted
version of the same function, but keep around the old function while
we transition. Adopt the Twisted version of the function when listing
reviewers of a PR.

* Tools/CISupport/ews-build/steps.py:
(GitHubMixin):
(GitHubMixin.fetch_data_from_url_with_authentication_github): Convert to Twisted.
(GitHubMixin.fetch_data_from_url_with_authentication_github_old): Retain old
synchronous version.
(GitHubMixin.get_pr_json): Use old synchronous version.
(GitHubMixin.get_reviewers): Use new Twisted version.
(GitHubMixin.remove_labels): Use old synchronous version.
(ValidateCommitterAndReviewer.run): Use Twisted asynchronous requests.
* Tools/CISupport/ews-build/steps_unittest.py:
(TestGitHubMixin.test_no_reviewers): Return defered to match API we're mocking.
(TestGitHubMixin.test_single_review): Ditto.
(TestGitHubMixin.test_multipe_reviews): Ditto.
(TestGitHubMixin.test_retracted_review): Ditto.
(TestGitHubMixin.test_pagination): Ditto.
(TestGitHubMixin.test_reviewers_invalid_response): Ditto.
(TestGitHubMixin.test_reviewers_error): Ditto.

Canonical link: https://commits.webkit.org/259427@main
